pkg/controller/command/command_controller.go

```diff
@@ -608,7 +602,13 @@ func (r *ReconcileCommand) listCommandsPods(commandName string) (*core.PodList,
 
 func (r *ReconcileCommand) prepareIntendedDeployment(instanceDeployment *apps.Deployment, commandCR *contrail.Command) error {
 	instanceDeploymentName := commandCR.Name + "-command-deployment"
-	intendedDeployment := contrail.SetDeploymentCommonConfiguration(instanceDeployment, &commandCR.Spec.CommonConfiguration)
+	podConfiguration := commandCR.Spec.CommonConfiguration
+	if commandCR.Status.UpgradeState == contrail.CommandShuttingDownBeforeUpgrade ||
```

Comment:
if isUpgrading() didn't contain upgrade failed state, you could use it here :)

Reply:
done